### PR TITLE
safer_getattr

### PR DESCRIPTION
**Pull request by loechel**
fix a possible breakout possibility in the provided safer_getattr methode in Guards

### CODE REVIEW

File: src/RestrictedPython/Guards.py [hunk 1]
```diff
@@ -246,6 +246,8 @@ def safer_getattr(object, name, default=None, getattr=getattr):
     http://lucumr.pocoo.org/2016/12/29/careful-with-str-format/
 
     """
+    if not (isinstance(name, str) and type(name) is str):
```
Comment (d-maurer) [inline src/RestrictedPython/Guards.py:249]:
```suggestion
    if type(name) is not str:
```

File: src/RestrictedPython/Guards.py [hunk 2]
```diff
@@ -246,6 +246,8 @@ def safer_getattr(object, name, default=None, getattr=getattr):
...
+        raise TypeError('name is not of type str')
```
Comment (d-maurer) [inline src/RestrictedPython/Guards.py:250]:
```suggestion
        raise TypeError('type(name) must be str')
```

File: CHANGES.rst [hunk 3]
```diff
@@ -9,6 +9,7 @@ Changes
 - Allow to use the package with Python 3.13 -- Caution: No security
   audit has been done so far.
 - Add support for single mode statements / execution.
+- Fix a potential breakout capability in the provided safer_getattr method that is part of the safer_builtins
```
Comment (icemac) [inline CHANGES.rst:12]:
```suggestion
- Fix a potential breakout capability in the provided ``safer_getattr`` method
  that is part of the ``safer_builtins``.
```

File: tests/test_Guards.py [hunk 4]
```diff
@@ -259,6 +259,42 @@ def test_Guards__safer_getattr__3():
...
+        __name__=None,
+        __metaclass__=type,
+        # _write_=_write_,
+        getattr=safer_getattr,
+        result=None,
+    )
+
+    with pytest.raises(TypeError) as err:
+        restricted_exec(SAFER_GETATTR_BREAKOUT, restricted_globals)
+    assert 'name is not of type str' == str(err.value)
```
Comment (icemac) [inline tests/test_Guards.py:295]:
```suggestion
    assert 'type(name) must be str' == str(err.value)
```

**Review by icemac — APPROVED** [review]

**Comment by icemac** [comment]
Released in https://pypi.org/project/RestrictedPython/7.2/

**Pull request merged**